ui/src/app/shared/model/detailed-app-registration.model.ts

```diff
@@ -0,0 +1,93 @@
+import { ApplicationType } from '../../shared/model/application-type';
+import { AppRegistration } from '../../shared/model/app-registration.model';
+import { Serializable } from '../../shared/model';
+
+/**
+ * The DetailedAppRegistration provides additional information compared
+ * to the base {@link AppRegistration} such as options
+ * ({@link ConfigurationMetadataProperty}).
+ *
+ * @author Gunnar Hillert
+ */
+export class DetailedAppRegistration extends AppRegistration implements Serializable<DetailedAppRegistration> {
+
+  public options: ConfigurationMetadataProperty[];
+
+  constructor(
+    name?: string,
+    type?: ApplicationType,
+    uri?: string ) {
+      super(name, type, uri);
+    }
+
+  /**
+   * For a given JSON data object, this method
+   * will populate the corresponding AppRegistration object, with
+   * the provided properties.
+   *
+   * @param input JSON input data
+   */
+  public deserialize(input) {
+    super.deserialize(input);
+
+    if (input.options) {
+      this.options = [];
+      for (const option of input.options) {
+        this.options.push(new ConfigurationMetadataProperty().deserialize(option));
+      }
+    }
+    return this;
+  }
+}
+
+export class ConfigurationMetadataProperty implements Serializable<ConfigurationMetadataProperty> {
+
+  public id: string;
+  public name: string;
+  public type: string;
+  public description: string;
+  public shortDescription: string;
+  public defaultValue: string;
+  public deprecation: Deprecation;
+  public isDeprecated: boolean;
+
+  /**
+   * For a given JSON data object, this method
+   * will populate the corresponding {@link ConfigurationMetadataProperty}
+   * object, with the provided properties.
+   *
+   * @param input JSON input data
+   */
+  public deserialize(input) {
+    this.id = input.id ? input.id : undefined;
+    this.name = input.name ? input.name : undefined;
+    this.type = input.type ? input.type : undefined;
+    this.description = input.description ? input.description : undefined;
+    this.shortDescription = input.shortDescription ? input.shortDescription : undefined;
+    this.defaultValue = input.defaultValue ? input.defaultValue : undefined;
+    this.deprecation = input.deprecation ? new Deprecation().deserialize(input.deprecation) : undefined;
+    this.isDeprecated = input.deprecated;
```

Comment:
nitpicky I know:  But shouldn't this be:    ` this.isDeprecated = input.deprecated ? input.deprecated : false;`  else if `deprecated` is missing it will show up as undefined.

Reply:
Yes - agree. False it is.